proposals/NNNN-vehicle-type-filter.md

```diff
@@ -0,0 +1,435 @@
+
+The exclusive apps should not host `SDL Router service` on unsupported SDL systems. The android system requires all apps using foreground services to show a notification. The SDL enabled android app can have two foreground services. The `SDL router service` is started by the android proxy and another foreground service will be started by an SDL enabled app.
+
+The `SDL router service` has no information on what head unit it is connecting with when a transport connection is made. Other SDL apps bind to available router service and use this connection. This behavior could lead to cases where a proprietary app creates an `SDL router service` when connected to an unsupported SDL enabled system. This started service will need to show a foreground notification with the app name, giving the user the impression that the proprietary app is working with an unsupported SDL enabled system.
+
+For example, A proprietary app of an OEM creates Router service and the mobile device is connected to another OEMs' SDL enabled IVI system. The user would perceive that the proprietary app is doing something in the background, when connected to SDL enabled IVI system from another OEM.
+
+Please refer below screenshots.
+
+- Notification 1: Created by OEM App.
+- Notification 2: Created by SDL Android SDK for `SDL Router Service`.
+
+__Note the app name being shown in both notifications. In the below picture, the `SDL App` is using the “SDL Router Service” created by the `OEM App`.__
+
+<img src="../assets/proposals/NNNN-vehicle-type-filter/two_noti.png" alt="App name in notifications" width="500"/>
+
+
+The only way to remove notifications from OEM App when it connects to another vehicle is to force stop both the services so, that both notifications are removed. In this case, other apps connected through the `OEM Apps'` router service will also get disconnected. This is not a recommended approach. If the OEM app wants to unregisters itself from the SDL enabled IVI system, it will stop the service created by an app. The `SDL router service` from the OEM app will keep running.
+
+Please refer below screenshot. The `SDL Router Service` is hosted by the OEM app, the `Notification 2` with the app name will be shown to the user, giving the impression that the `OEM App` is still working. If the OEM App starts showing notifications when connected to another OEM's vehicle, this would confuse users. They would probably think that `OEM App` on their mobile device is working with another OEMs SDL enabled IVI system. This proposal will define ways to mitigate this issue.
+
+<img src="../assets/proposals/NNNN-vehicle-type-filter/one_noti.png" alt="App name in SDL router service notifications" width="500"/>
+
+This proposal tries to address the router service notification issue by defining the vehicle type filter. So, we could have an app that could specifically connect to certain vehicles. For example, an App that is designed only for Mustang vehicles.
```

Comment:
Can you please update references to the router service for clarity?  If you're referring to the service/code itself in the Java Suite library, please use "`SdlRouterService.java`".  Otherwise, you can use "Trusted Router Service" (without quotes or code blocks) to reference the idea of the router service.  More information can be found here: https://smartdevicelink.com/en/guides/android/frequently-asked-questions/trusted-router-service/.

Reply:
Hi @theresalech, I am referring to the ["Android Router Service"](https://smartdevicelink.com/en/guides/android/frequently-asked-questions/trusted-router-service/#what-is-the-android-router-service) described in the link you shared. I wanted to refer to a service that creates a notification on the mobile devices. I guess this can be trusted or untrusted. Please let me know, how should we refer to service that is creating  "Connected to Ford Focus" notification shown in the picture below.
![one_noti](https://user-images.githubusercontent.com/24958343/74734742-75c59b00-524f-11ea-9eec-cebb3b7c93b2.png)

Reply:
Thanks, @ashwink11!  As noted in the link I shared, "it is important that the app hosting the router service can be trusted. This is done through a certification process and a back-end server that maintains a database of apps that can act as a Trusted Router Service."  Therefore, please update to "Trusted Router Service" (without quotes or code blocks) when talking about the general concept, and use "`SdlRouterService.java`" if/when you're talking about the specific code.  Code blocks around "Trusted Router Service" as a concept are not necessary, and it's helpful for those reading the proposal to have consistent capitalization throughout the proposal, so please fix `SDL Router service`, `SDL Router Service`, `SDL router service`, etc. to use either "Trusted Router Service" or `SdlRouterService.java` based on the context of the reference.

Reply:
Thank you for the info @theresalech. The notification we see in the screenshot shared in the above comment was not from a trusted SDLC certified app. They were just some sample apps with random App name and id. Still, these apps were creating router service.

Please find the highlighted text in the below picture.
![Capture](https://user-images.githubusercontent.com/24958343/74818304-af091400-52fe-11ea-9e5b-4455a01dc16f.PNG)
I understand form the highlighted text that Router service can be trusted or untrusted. 
In this proposal, I want to refer to the general concept of the Android router service, it does not matter if they are trusted or not.

Could you please confirm that if we use the term "Trusted Router Service" for the general concept, it would refer to router services from all apps, both certified and uncertified?

Reply:
@ashwink11 I believe you can use "Android Router Service" if you're referencing both certified and uncertified.  This may be a topic for Steering Committee discussion once the proposal is actually under review.  I'm simply trying to ensure clarity prior to that point.  Please let me know if you have any more questions, thanks!

Reply:
Thank you @theresalech. I updated the proposal.